lib/screens/v2/disruption_diagram/model.ex

```diff
@@ -0,0 +1,71 @@
+defmodule Screens.V2.DisruptionDiagram.Model do
+  @moduledoc """
+  Struct and functions to generate and model a disruption diagram.
+  """
+
+  # Model fields TBD
+  defstruct []
+
+  @type t :: %__MODULE__{}
+
+  @type serialized_response :: continuous_disruption_diagram() | discrete_disruption_diagram()
+
+  @type continuous_disruption_diagram :: %{
+          effect: :shuttle | :suspension,
+          # A 2-element list, giving indices of the effect region's *first and last disrupted stops*.
+          # For example in this scenario:
+          #     0     1     2     3     4     5     6     7     8
+          #    <= === O ========= O - - X - - X - - X - - O === =>
+          #                             |---range---|
+          # The range is [4, 6].
```

Comment:
So sorry for my delayed review! The way this alert would be crafted by OIO's, and the way the alert text will read, is "Shuttling between Station 3 and Station 7". And the range or "emphasis" label on maps will also cover the same [area](https://www.figma.com/file/l4AZl8UXyKLAFI8OtultQk/Pre-Fare-%E2%80%94-Alerts-V2?type=design&node-id=565-11629&mode=dev). But I do kinda see the benefit of doing it this way, so that determining the icons for the stations are easier? What do you think?

Reply:
Yeah, it's a little weird but the value of the range is purely based on visuals, not the semantics of the different types of alerts. It's simply "which are the first and last slots that should get disruption-style symbols?"

That lets the client focus on presentation rather than having to know more details about how the different alerts should be interpreted.

Also—if the shuttle replacement/suspension starts or ends at a terminus, then there would be no way to represent that in the data if we had the range start/end at the first/last "non-disrupted" stops.

Reply:
Aha! Sweet, thanks for the extra explanation. This is perfect

Reply:
Both of you flagged this and you were totally right! The range does need to include the "boundary" stops. And, my reason for taking this approach originally:

> if the shuttle replacement/suspension starts or ends at a terminal, then there would be no way to represent that in the data if we had the range start/end at the first/last "non-disrupted" stops. (We couldn't get that terminal stop to have a disruption-style symbol.)

can be worked around with a pretty straightforward check on the client: "If the start of the range is 0 use a 'disruption icon' for the starting border stop. If the end of the range is (length(diagram) - 1) use a 'disruption icon' for the ending border stop."

I'll update the comment now.